UI/src/model/selectedLaneModelling.js

```diff
@@ -0,0 +1,588 @@
+          initial_road_part_number: initial_road_part_number,
+          initial_distance: initial_distance,
+          end_road_part_number: parseInt(end_road_part_number),
+          end_distance: parseInt(end_distance),
+          track: track,
+          lanes: lanes
+        };
+      }else{
+        payload = {
+          linkIds: linkIds,
+          sideCode: sideCode,
+          lanes: lanes.concat(omitUnrelevantProperties(assetsToBeExpired)).concat(omitUnrelevantProperties(assetsToBeRemoved))
+        };
+      }
+
+      var backendOperation = isAddByRoadAddressActive ? backend.updateLaneAssetsByRoadAddress : backend.updateLaneAssets;
+
+      backendOperation(payload, function() {
+        dirty = false;
+        self.close();
+        eventbus.trigger(singleElementEvent('saved'));
+      }, function(error) {
+        jQuery('.spinner-overlay').remove();
+        alert(error.responseText);
+      });
+    };
+
+    var cancelExisting = function() {
+      selection = lanesFetched;
+      dirty = false;
+      eventbus.trigger(singleElementEvent('valueChanged'), self);
+    };
+
+    this.cancel = function() {
+      cancelExisting();
+      self.close();
+      eventbus.trigger(singleElementEvent('cancelled'), self);
+    };
+
+    this.verify = function() {
+      eventbus.trigger(singleElementEvent('saving'));
+      var knownLinearAssets = _.reject(selection, isUnknown);
+      var payload = {ids: _.map(knownLinearAssets, 'id'), typeId: typeId};
+      collection.verifyLinearAssets(payload);
+      dirty = false;
+      self.close();
+    };
+
+    this.exists = function() {
+      return !_.isEmpty(selection);
+    };
+
+    var getProperty = function(lane, propertyName) {
+      return _.has(lane, propertyName) ? lane[propertyName] : null;
+    };
+
+    this.getId = function() {
+      return getProperty('id');
+    };
+
+    this.setEndAddressesValues = function(currentPropertyValue) {
+      var endValue = _.head(currentPropertyValue.values);
+      switch(currentPropertyValue.publicId) {
+        case "end_road_part_number":
+          end_road_part_number = _.isEmpty(endValue) ? endValue : endValue.value;
+          break;
+        case "end_distance":
+          end_distance = _.isEmpty(endValue) ? endValue : endValue.value;
+          break;
+      }
+
+      _.forEach(selection, function (lane) {
+        var currentLaneNumber = _.head(_.find(lane.properties,function (prop) {
+          return prop.publicId == "lane_code";
+        }).values).value;
+
+        var properties = _.filter(self.getValue(currentLaneNumber), function(property){ return property.publicId !== currentPropertyValue.publicId; });
+        properties.push(currentPropertyValue);
+        self.setValue(currentLaneNumber, {properties: properties});
+      });
+    };
+
+    this.getValue = function(laneNumber, marker) {
+      var lane = self.getLane(laneNumber, marker);
+
+      var value = getProperty(lane, 'properties');
+      return value;
+    };
```

Comment:
this can be reduced to:
```JavaScript
this.getValue = function(laneNumber, marker) {
      return getProperty(self.getLane(laneNumber, marker), 'properties');
    };

Reply:
Done